### PR TITLE
100 login action reducer

### LINKED ISSUE #100: Write an action and a reducer for login
Write an action and a corresponding reducer for log in that can be called from a component.

Please see https://github.com/kowhai-2018/Final-Project/blob/development/client/actions/auth.js and https://github.com/kowhai-2018/Final-Project/blob/development/client/reducer/auth.js for example code.

Please note that in the example code the api and the action are in the same function but these should be separated. The api for handling the login will be in a separate api file, see issue #95.

### PR DESCRIPTION
**Pull request by ari-t**
closes #100 

This is relying on the API from issue #95

### CODE REVIEW

File: client/actions/auth.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export const signinError = error => {
+  return {
+    type: 'SIGNIN_ERROR',
+    error
+  }
+}
+
+// user = {name, surname, password}
```
Comment (emilyparkes) [inline client/actions/auth.js]: remove comments please

File: client/reducers/auth.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import {getToken} from '../utils/tokens'
+
+const initialState = {
+  error: null,
+  loggedIn: !!getToken(), // Coerces object to boolean, so either returns true or false
```
Comment (emilyparkes) [inline client/reducers/auth.js]: remove comments here too please

File: client/reducers/auth.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  newRegistration: false,
+  pending: false
+}
+
+export default function authReducer (state = initialState, action) {
+  switch (action.type) {
+    case 'SIGNIN_PENDING':
```
Comment (emilyparkes) [inline client/reducers/auth.js:12]: what does this return?

File: client/reducers/auth.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      }
+    case 'REGISTER_SUCCESS':
+      return {
+        ...state,
+        error: null,
+        newRegistration: true,
+        loggedIn: true,
+        pending: false
+      }
+    case 'SIGNIN_ERROR':
```
Comment (emilyparkes) [inline client/reducers/auth.js:34]: what does this return?

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: client/reducers/auth.js (hunk 3 above)
Comment (ari-t) [reply client/reducers/auth.js:12]: We can use multiple cases one after the other  which use the same logic.
case 'SIGNIN_PENDING':
case 'REGISTER_PENDING':
      return ...
would be logically the same as 
case 'SIGNIN_PENDING' || 'REGISTER_PENDING':
      return ...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

**Review by emilyparkes — CHANGES_REQUESTED** [review]
please remove comments and pull from dev, please also check whether these actions once called change the state in redux dev tools as you expect

**Review by emilyparkes — APPROVED** [review]

**Pull request merged**